openedx/openedx-platform
# Support Zip files for Course Import

## Description

Support `zip` files for Course import.

Please see more details at this issue: https://github.com/openedx/platform-roadmap/issues/285

Useful information to include:
- Which edX user roles will this change impact?
=> "Course Author".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
=> Before
<img width="908" alt="image" src="https://github.com/openedx/edx-platform/assets/131209/bb7fda65-2a94-4222-b7d4-bfcf42a8b2cc">
=> After
<img width="917" alt="image" src="https://github.com/openedx/edx-platform/assets/131209/3760cc51-486b-4d2c-a453-10d8f23664e7">
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these
changes.
=> None

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.
https://tasks.opencraft.com/browse/BB-8025
https://github.com/openedx/platform-roadmap/issues/285

## Testing instructions
* Select the course
  1. Check out this PR
  2. Open the CMS (Studio)
  3. Create a course or use the example one
  4. Select the course
  5. In the top menu, select `Tools / Export`
  6. Click in `Export Course Content`
  7.  Save the `.tar.gz` file in a new folder at your computer
* Modify the course
  1. Find this file in your computer
  2. Extract the course
  3. Modify the course content - Optional
  4. Compact the course again in `.zip` file
  5. Rename the compressed file (can have spaces in the name) - Optional
* Upload changes
  1. Open the CMS (Studio)
  2. Select the course
  3. In the top menu, select `Tools / Import`
  4. Click in `Choose new file`
  5. Select the created `.zip` file
  6. Click in `Replace my course with the selected file`
  7. Wait the import steps to finish
  8. Check if changes are persisted - Optional

## Deadline
=> 2023-10-29

## Other information
Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
=> No
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.
=> No
- If your [database migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) can't be rolled back easily.
=> No
